fedimint-server/src/config/api.rs

```diff
@@ -511,66 +509,52 @@ impl ConfigGenApi {
-        let client = create_bitcoind(bitcoin_rpc_config).map_err(|e| {
+    async fn fetch_bitcoin_status() -> ApiResult<BitcoinRpcConnectionStatus> {
+        let bitcoin_rpc_config = BitcoinRpcConfig::get_defaults_from_env_vars().map_err(|e| {
+            ApiError::server_error(format!("Failed to get bitcoin rpc env vars: {e}"))
+        })?;
+        let client = create_bitcoind(&bitcoin_rpc_config).map_err(|e| {
             ApiError::server_error(format!("Failed to connect to bitcoin rpc: {e}"))
         })?;
-        // TODO: Find a better way to check if esplora is synced, just returning Synced
-        // if it connects
-        let _network_info = client
-            .get_network()
+        let block_count = client.get_block_count().await.map_err(|e| {
+            ApiError::server_error(format!("Failed to get block count from bitcoin rpc: {e}"))
+        })?;
+        let chain_tip_block_height = block_count - 1;
+        let chain_tip_block_hash = client
+            .get_block_hash(chain_tip_block_height)
             .await
-            .map_err(|e| ApiError::server_error(format!("Failed to get esplora info: {e}")))?;
-        Ok(BitcoinRpcConnectionStatus::Synced)
-    }
-}
+            .map_err(|e| {
+                ApiError::server_error(format!(
+                    "Failed to get block hash for block count {block_count} from bitcoin rpc: {e}"
+                ))
+            })?;
+        let chain_tip_block = client.get_block(&chain_tip_block_hash).await.map_err(|e| {
+            ApiError::server_error(format!(
+                "Failed to get block for block hash {chain_tip_block_hash} from bitcoin rpc: {e}"
+            ))
+        })?;
+        let chain_tip_block_time = chain_tip_block.header.time;
 
-fn check_bitcoind_status(
-    bitcoin_rpc_config: &BitcoinRpcConfig,
-) -> ApiResult<BitcoinRpcConnectionStatus> {
-    let (url, auth) = fedimint_bitcoind::bitcoincore::from_url_to_url_auth(&bitcoin_rpc_config.url)
-        .map_err(|e| ApiError::server_error(format!("Failed to parse bitcoin rpc url: {e}")))?;
-    let client = bitcoincore_rpc::Client::new(&url, auth)
-        .map_err(|e| ApiError::server_error(format!("Failed to connect to bitcoin rpc: {e}")))?;
-    let blockchain_info = block_in_place(|| client.get_blockchain_info())
-        .map_err(|e| ApiError::server_error(format!("Failed to get blockchain info: {e}")))?;
-
-    if blockchain_info.initial_block_download {
-        return Ok(BitcoinRpcConnectionStatus::Syncing(
-            blockchain_info.verification_progress,
-        ));
+        Ok(BitcoinRpcConnectionStatus {
+            chain_tip_block_height,
+            chain_tip_block_time,
+        })
     }
-
-    Ok(BitcoinRpcConnectionStatus::Synced)
 }
 
 #[derive(Debug, Serialize, Deserialize, Clone, Copy)]
-pub enum BitcoinRpcConnectionStatus {
-    Synced,
-    Syncing(f64),
+pub struct BitcoinRpcConnectionStatus {
+    chain_tip_block_height: u64,
+    chain_tip_block_time: u32,
```

Comment:
Useful to make it a `SystemTime`?

Reply:
I don't see the purpose of that if we're just going to serialize it afterwards. Can't the API caller convert it in whatever language it uses?

Reply:
I guess it's enough if only passed through. Generally typing things is better though.